src/scripts/main.js

```diff
@@ -21,7 +20,19 @@
 const tree = document.querySelector('#tree');
 
 function createTree(element, data) {
-  // WRITE YOUR CODE HERE
+  const ul = document.createElement('ul');
+
+  for (let key in data) {
+    const li = document.createElement('li');
+    li.textContent = key;
+    ul.appendChild(li);
+
+    if (typeof data[key] === 'object' && Object.keys(data[key]).length > 0) {
+      li.appendChild(createTree(li, data[key]));
+    }
```

Comment:
You are checking if `data[key]` is an object and if it has any keys before appending it to the list. This is a good practice as it prevents unnecessary function calls and DOM operations. However, the condition `typeof data[key] === 'object'` will also return true for arrays, functions, and null. If you want to ensure that `data[key]` is a plain object, you can use `Object.prototype.toString.call(data[key]) === '[object Object]'` instead.